src/node/routes/vscode.ts

```diff
@@ -33,27 +34,43 @@ export class CodeServerRouteWrapper {
       })
     }
 
-    const { query } = await req.settings.read()
-    if (query) {
-      // Ew means the workspace was closed so clear the last folder/workspace.
-      if (req.query.ew) {
-        delete query.folder
-        delete query.workspace
-      }
+    let folder = undefined
+    let workspace = undefined
+    const to = self(req)
+    const settings = await req.settings.read()
+    const lastOpened = settings.query || {}
+
+    // Ew means the workspace was closed so clear the last folder/workspace.
+    if (req.query.ew) {
+      delete lastOpened.folder
+      delete lastOpened.workspace
+    }
```

Comment:
I suppose we could move this into the `if` block below as well.

Reply:
Actually it occurs to me we only need to read settings when there are no query parameters too, which could be a bit of an optimization.